examples/MMPT/demo_sign.py

```diff
@@ -97,73 +97,57 @@ def preprocess_text(text, model_name='default'):
     return caps, cmasks
 
 
-def embed_pose(pose, model_name='default'):
-    model = models[model_name]['model']
-
-    caps, cmasks = preprocess_text('', model_name)
-    poses = pose if type(pose) == list else [pose]
-    embeddings = []
-
-    for pose in poses:
-        pose_frames = preprocess_pose(pose)
-
-        with torch.no_grad():
-            output = model(pose_frames, caps, cmasks, return_score=False)
-            embeddings.append(output['pooled_video'].cpu().numpy())
-
-    return np.concatenate(embeddings)
-
+def score_pose_and_text(pose, text, max_frames, model_name="default"):
+    model = models[model_name]["model"]
 
-def embed_text(text, model_name='default'):
-    model = models[model_name]['model']
-
-    # pose_frames = torch.randn(1, 1, 534)
-    pose_frames = torch.randn(1, 1, 609)
-    texts = text if type(text) == list else [text]
-    embeddings = []
-
-    for text in texts:
-        caps, cmasks = preprocess_text(text, model_name)
-
-        with torch.no_grad():
-            output = model(pose_frames, caps, cmasks, return_score=False)
-            embeddings.append(output['pooled_text'].cpu().numpy())
-
-    return np.concatenate(embeddings)
-
-
-def score_pose_and_text(pose, text, model_name='default'):
-    model = models[model_name]['model']
-
-    pose_frames = preprocess_pose(pose)
+    pose_frames = preprocess_pose(pose, max_frames)
     caps, cmasks = preprocess_text(text)
 
     with torch.no_grad():
         output = model(pose_frames, caps, cmasks, return_score=True)
-    
+
     return text, float(output["score"])  # dot-product
 
 
-def score_pose_and_text_batch(pose, text, model_name='default'):
-    pose_embedding = embed_pose(pose, model_name)
-    text_embedding = embed_text(text, model_name)
+def main():
+    parser = argparse.ArgumentParser(description="Evaluate pose and text similarity using SignCLIP.")
```

Comment:
I wrote it rather quickly without planning to make it a CLI tool, but thanks for adding the ArgumentParser ;)

Reply:
could possibly refactor it to allow arbitrary text input

Reply:
You're welcome! I actually have another version that I used to create and save .npy embeddings, on which some of these changes are based

Reply:
It let me point the demo to a folder, specify which models to load, etc. https://github.com/cleong110/semantic-sign-language-search/blob/main/setup_signCLIP/demo_sign_embed_and_save.py